scipy/special/__init__.py

```diff
@@ -861,3 +861,35 @@ def _get_include():
     """
     import os
     return os.path.dirname(__file__)
+
+
+def _load_libsf_error_state():
+    """Load libsf_error_state.dll shared library on Windows
+
+    libsf_error_state manages shared state used by
+    ``scipy.special.seterr`` and ``scipy.special.geterr`` so that these
+    can work consistently between special functions provided by different
+    extension modules. This shared library is installed in scipy/special
+    alongside this __init__.py file. Due to lack of rpath support, Windows
+    cannot find shared libraries installed within wheels. To circumvent this,
+    we directly load lib_sf_error_state.dll when on Windows.
+
+    The logic for this function was borrowed from the function ``make__init``
+    in `scipy/tools/openblas_support.py`.
+
+    https://github.com/scipy/scipy/blob/bb92c8014e21052e7dde67a76b28214dd1dcb94a/tools/openblas_support.py#L239-L274
+    """  # noqa: E501
+    import os
+    if os.name == "nt":
+        try:
+            from ctypes import WinDLL
+            basedir = os.path.dirname(__file__)
+        except:
+            pass
```

Comment:
The linter is displeased with the bare except. I'm not sure what the try/except block is for. I just copied it from:

https://github.com/scipy/scipy/blob/bb92c8014e21052e7dde67a76b28214dd1dcb94a/tools/openblas_support.py#L252-L258

Is it actually necessary? I can throw in a `noqa`. I suppose it has something to do with the comment

```python
 # convention for storing / loading the DLL from
                # numpy/.libs/, if present
```

the linter is displeased with the bare except. I could add `no

Reply:
yes, good you put it back. this is quite fragile and I can't quite remember what can go wrong; prefer to change as little as possible.